### ISSUE TITLE
Need a way to specify where a shared window or camera should appear

### ISSUE THREAD

**Issue by blairmacintyre** [mozilla/Spoke#929]
We are going to be using hubs rooms for giving talks.  A speaker will want to have slides, video and other assets:  those can be pre-set.

But, they will also want to share the video of their camera, or a window on their computer.  Right now, that has to be done manually each time they connect, and if they lose connection part way through, they need to tediously share again and reposition / resize when they reconnect.

We really need a way to lay out a space in a scene that acts as the default location for shared screens/cameras.   The simple initial version could simply say "first shared item here", "second shared item here", etc.  

A simple implementation might use named images:  I upload an image, name it "shared-media-slot-1", and hubs will position and size the 1st share into the room the same size as that image, slightly in front (to avoid z-fighting).

I would mark this URGENT  for VR, if we can get ANY working way rigged up.

**Comment by gfodor** [comment]
This is the media zones feature (can’t link atm) - I believe it is too
close to expect implementation of this in time for conference, too risk
laden. Defer to others tho if they feel confident.

**Comment by gfodor** [comment]
Also unless you plan on doing a full dry run with all speakers, I’d advise
against focusing on in-room talks vs pre-recorded - most likely talks will
fail due to speakers not having familiarity with the tool unless they have
done a full dry run. (Maybe this is the plan, but I am out of the loop obv!)

**Comment by blairmacintyre** [comment]
(We are working through plans and options, but a sizable number of them across different events at the conference, come back to needing screen sharing, and having it be "pre-configurable" so it's not "half-assed" during the necessary time.  So, let's set aside the logistics of the conference for this discussion).

> This is the media zones feature (can’t link atm) - I believe it is too close to expect implementation of this in time for conference, too risk laden. Defer to others tho if they feel confident.

I remember some discussions, didn't know where they went.  But, this is much less than the full, general feature.  Since the client already has to pick a place for media, I was trying to imagine a very simple version that would work;  i.e., when the scene loads, if there are images/videos/whatever with the correct name, simply use their position/orientation/size as the location for the thing.  Right now, it just drops things in front of the person sharing.

I could imagine also supporting media drops, etc.  But, that's less urgent;  those things can be adjusted and pinned ahead of time.

**Comment by gfodor** [comment]
I’d advise a hard coded fork hack for this in the case it’s critical -
anything generalized, even if dark code, adds some risk insofar as it sits
on the main media spawning code path. Assuming permissions are locked down
so that only presenters can spawn, having a scene-specific conditional to
do placement seems relatively safe.

**Comment by lukevink** [comment]
With corona and all, we elected to use Hubs to build our entire office in it, room by room, so we could attempt to use it both for social fun, but also for our large meetings and presentations. (50-70 people). This worked really well - except, we couldnt position the shared screen correctly in all of the chaos, and had to switch to google hangouts :(.

It would be really amazing if we could elect to put our shard screen on a video object for example, so we could have a fixed place in the room that people could share their screen to for presentations. Is this what you mean by Media Zones?

**Comment by blairmacintyre** [comment]
yes, I think this is one of the ideas behind media zones.  I'd personally like it if we could have the system remember where a screen was last positioned when _I_ share it in a space, so it offers to put it back there for me when I share in the future.
